plugin-maven/src/main/java/com/diffplug/spotless/maven/typescript/Tsfmt.java

```diff
@@ -0,0 +1,105 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.typescript;
+
+import java.io.File;
+import java.util.Map;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.maven.FormatterStepConfig;
+import com.diffplug.spotless.maven.FormatterStepFactory;
+import com.diffplug.spotless.npm.TsConfigFileType;
+import com.diffplug.spotless.npm.TsFmtFormatterStep;
+import com.diffplug.spotless.npm.TypedTsFmtConfigFile;
+
+public class Tsfmt implements FormatterStepFactory {
+
+	@Parameter
+	private String tslintFile;
+
+	@Parameter
+	private String tsconfigFile;
+
+	@Parameter
+	private String vscodeFile;
+
+	@Parameter
+	private String tsfmtFile;
+
+	@Parameter
+	private String typescriptFormatterVersion;
+
+	@Parameter
+	private String typescriptVersion;
+
+	@Parameter
+	private String tslintVersion;
+
+	@Parameter
+	private String npmExecutable;
+
+	@Parameter
+	private Map<String, Object> config;
+
+	@Parameter(defaultValue = "${project.build.directory}", required = true, readonly = true)
+	private File buildDir;
+
+	@Override
+	public FormatterStep newFormatterStep(FormatterStepConfig stepConfig) {
+
+		Map<String, String> devDependencies = TsFmtFormatterStep.defaultDevDependencies();
+		if (typescriptFormatterVersion != null) {
+			devDependencies.put("typescript-formatter", typescriptFormatterVersion);
+		}
+		if (typescriptVersion != null) {
+			devDependencies.put("typescript", typescriptVersion);
+		}
+		if (tslintVersion != null) {
+			devDependencies.put("tslint", tslintVersion);
+		}
+
+		File npm = npmExecutable != null ? stepConfig.getFileLocator().locateFile(npmExecutable) : null;
+
+		TypedTsFmtConfigFile configFile = null;
+
+		// check that there is only 1 config file or inline config
+		if (this.tsconfigFile != null
+				^ this.tsfmtFile != null
+				^ this.tslintFile != null
+				^ this.vscodeFile != null) {
+			if (this.tsconfigFile != null) {
+				configFile = new TypedTsFmtConfigFile(TsConfigFileType.TSCONFIG, stepConfig.getFileLocator().locateFile(tsconfigFile));
+			} else if (this.tsfmtFile != null) {
+				configFile = new TypedTsFmtConfigFile(TsConfigFileType.TSFMT, stepConfig.getFileLocator().locateFile(tsfmtFile));
+			} else if (this.tslintFile != null) {
+				configFile = new TypedTsFmtConfigFile(TsConfigFileType.TSLINT, stepConfig.getFileLocator().locateFile(tslintFile));
+			} else if (this.vscodeFile != null) {
+				configFile = new TypedTsFmtConfigFile(TsConfigFileType.VSCODE, stepConfig.getFileLocator().locateFile(vscodeFile));
+			}
+		} else {
+			if (config == null) {
+				throw new IllegalArgumentException("must specify exactly one configFile or config");
+			}
+		}
+
+		if (buildDir == null) {
+			buildDir = new File(".");
+		}
```

Comment:
... because if it was, this wouldn't be necessary.  I think part of what might be happening is that new File(".") is not resolving to where we suspect it to be.  We could modify `FormatterStepConfig` to include this value.  It already has a `FileLocator`, but unfortunately that can only be used to get files.  Should be able to add the base directory to this, I imagine....

Reply:
Hmm... I tried this below and it is still not working.